cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/sql/GcpDatabaseServerUpgradeService.java

```diff
@@ -1,21 +1,96 @@
 public class GcpDatabaseServerUpgradeService extends GcpDatabaseServerBaseService implements DatabaseServerUpgradeService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GcpDatabaseServerUpgradeService.class);
+
+    @Inject
+    private DatabasePollerService databasePollerService;
+
+    @Inject
+    private GcpSQLAdminFactory gcpSQLAdminFactory;
+
+    @Inject
+    private GcpStackUtil gcpStackUtil;
+
     @Override
     public List<CloudResource> upgrade(AuthenticatedContext ac, DatabaseStack stack, PersistenceNotifier resourceNotifier, TargetMajorVersion databaseVersion)
             throws Exception {
+        GcpDatabaseServerView databaseServerView = new GcpDatabaseServerView(stack.getDatabaseServer());
+        String deploymentName = databaseServerView.getDbServerName();
+        SQLAdmin sqlAdmin = gcpSQLAdminFactory.buildSQLAdmin(ac.getCloudCredential(), ac.getCloudCredential().getName());
+        String projectId = gcpStackUtil.getProjectId(ac.getCloudCredential());
+        String availabilityZone = ac.getCloudContext().getLocation().getAvailabilityZone().value();
+        List<CloudResource> buildableResource = new ArrayList<>();
+        buildableResource.add(getGcpDatabase(deploymentName, availabilityZone));
+
+        try {
+            InstancesListResponse list = sqlAdmin.instances().list(projectId).execute();
+            Optional<DatabaseInstance> first = Optional.empty();
+            if (!list.isEmpty()) {
+                first = list.getItems()
+                        .stream()
+                        .filter(e -> e.getName().equals(deploymentName))
+                        .findFirst();
+            }
+            if (!first.isEmpty()) {
+                DatabaseInstance databaseInstance = first.get();
+                databaseInstance.setDatabaseVersion(databaseServerView.getDatabaseType() + "_" + databaseVersion.getVersion());
+                SQLAdmin.Instances.Patch patch = sqlAdmin.instances().patch(projectId, databaseInstance.getName(), databaseInstance);
+                patch.setPrettyPrint(Boolean.TRUE);
+                try {
+                    Operation operation = patch.execute();
+                    verifyOperation(operation, buildableResource);
+                    CloudResource operationAwareCloudResource = createOperationAwareCloudResource(buildableResource.get(0), operation);
+                    databasePollerService.upgradeDatabasePoller(ac, List.of(operationAwareCloudResource));
+                    buildableResource.forEach(dbr -> resourceNotifier.notifyUpdate(dbr, ac.getCloudContext()));
+                    return Collections.singletonList(operationAwareCloudResource);
+                } catch (GoogleJsonResponseException e) {
+                    throw new GcpResourceException(checkException(e), resourceType(), buildableResource.get(0).getName());
+                }
+            } else {
+                LOGGER.debug("Deployment does not exist: {}", deploymentName);
```

Comment:
this should notify the caller with an exception and cause the upgrade flow to fail, doesn't it?

Reply:
I would throw an exception to inform the user that upgrade was not possible due to this.

Reply:
Fixed: will throw GcpResourceException if deployment does not exist.